src/pollution_model/run_medusa_2.py

```diff
@@ -426,8 +433,55 @@ def store_pollution_model_in_database(engine: Engine, results: gpd.GeoDataFrame,
         The id of the current medusa2.0 model run, to associate with the results.
     """
     results["scenario_id"] = scenario_id
-    results.set_index("Index", inplace=True)
-    results.to_sql("medusa2_model_output", engine, if_exists="append", index=True)
+    results.index.names = ["spatial_feature_id"]
+    results.to_sql(MEDUSA2ModelOutput.__tablename__, engine, if_exists="append", index=True)
+
+
+def serve_pollution_model() -> None:
+    """Serve the geospatial data for pollution models visualisation and API use.
+    Joins the medusa2_model_output table to the corresponding geospatial feature tables.
+    """
+    # Create geoserver workspace for pollution data
+    db_name = EnvVariable.POSTGRES_DB
+    workspace_name = f"{db_name}-pollution"
+    geoserver.create_workspace_if_not_exists(workspace_name)
+
+    # Ensure workspace has access to database
+    data_store_name = f"{db_name} PostGIS"
+    geoserver.create_db_store_if_not_exists(db_name, workspace_name, data_store_name)
+
+    # Serve medusa2_model_output joined to geometry from associated spatial table
+    medusa_output_layer_name = "medusa2_model_output"
+    pollution_sql_xml_query = rf"""
+        <metadata>
+          <entry key="JDBC_VIRTUAL_TABLE">
+            <virtualTable>
+              <name>{medusa_output_layer_name}</name>
+              <sql>
+                SELECT medusa2_model_output.*, geometry&#xd;
+                FROM medusa2_model_output&#xd;
+                INNER JOIN nz_roads&#xd;
+                ON spatial_feature_id = road_id&#xd;
+                WHERE surface_type = &apos;Rd&apos;&#xd;
+                UNION&#xd;
+                SELECT medusa2_model_output.*, geometry&#xd;
+                FROM medusa2_model_output&#xd;
+                INNER JOIN nz_building_outlines&#xd;
+                ON spatial_feature_id = nz_building_outlines.building_id&#xd;
+                WHERE surface_type &lt;&gt; &apos;Rd&apos;
+              </sql>
```

Comment:
this SQL escaping could be improved by inserting it as an XML-encoded string template.

Then we can write normal SQL and encode it similar to the example below:

```python
from xml.sax.saxutils import escape

# Original string with special characters
original_string = """SELECT * FROM table WHERE name = 'John' AND status <>'active'"""

# Encode the string with XML entity encoding
encoded_string = escape(original_string)

print(encoded_string)
>>>SELECT * FROM table WHERE name = &apos;John&apos; AND status &lt;&gt;&apos;active&apos;
```

Reply:
@LukeParky Let's go with this :)